File: exchange-bff/src/currency/model/exchange-price.ts
export class ExchangePrice {
    readonly from: string
    readonly to: string
    readonly value: number

    constructor(from: string, to: string, value: number) { 
        this.from = from
        this.to = to
        this.value = value
    }
}

export interface ExchangePrice {
    readonly from: string
    readonly to: string
    readonly value: number
}

File: forex-ms/src/main/kotlin/model/ExchangePrice.kt
import java.math.BigDecimal
import java.math.RoundingMode

data class ExchangePrice(
    val from: String,
    val to: String,
    val value: BigDecimal
) {
    constructor(from: Currency, to: Currency) : this(
        from.symbol,
        to.symbol,
        from.value.divide(to.value, 2, RoundingMode.HALF_UP)
    )
}
